user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/Configuring_components/Template_Editor.md

```diff
@@ -0,0 +1,338 @@
+
+> [!IMPORTANT]
+> Any changes made in the Template Editor are only saved once you click the *Save* button in the lower right corner. When you close the Template Editor by clicking the "X" in the top-right corner or the *Cancel* button in the lower right corner, you will be alerted of any unsaved changes that may be lost if you close the editor prematurely.
+
+## Working with layers
+
+In the side pane of the Template Editor, the *Layers* tab offers an overview of all layers. Layers are used in the Template Editor to make non-destructive edits by stacking icons, text, rectangles, and ellipses one on top of the other without compromising the pixels of individual elements.
+
+The benefits of working with layers:
+
+- You can add elements on top of each other without removing data.
+
+- You can apply conditions to only one layer of the template. See [Adding conditional cases to a layer](#adding-conditional-cases-to-a-layer).
+
+- You can discard any unsatisfactory layers without affecting the entire template.
+
+- You can mask layers so that they are temporarily hidden without the need to delete them.
+
+- ...
+
+If one layer is positioned higher than another, the higher layer will appear in front of the lower one. To **change the order of layers**, click the ![*Drag-and-drop*](~/user-guide/images/drag-and-drop.png) button and drag the layer up or down in the *Layers* tab. Release the mouse button when you reach the position where you want to place the selected layer.
+
+![Change order layers](~/user-guide/images/Change_Order_Layers.gif)<br/>*The Template Editor in DataMiner 10.4.1*
+
+To **mask a layer**, hover the mouse pointer over the layer and click the *eye* icon in the lower right corner. The layer is now no longer visible in the template's end result. Clicking the *eye* icon once more makes the layer visible again. Alternatively, you can mask the layer by adjusting its properties in the *Settings* pane. See [Specifying layer properties](#specifying-layer-properties).
+
+> [!NOTE]
+> In cases where you have masked a layer in both the *Layer* tab and the layer properties, undoing the masking in only one of these locations will result in the layer remaining invisible. Ensure that you reverse the masking in both places to render the layer visible again.
+
+To **delete a layer**, hover the mouse pointer over the layer and click the *delete* button in the top-right corner. Alternatively, select the associated tool (i.e. icon, text, rectangle, or ellipse) in the preview and press the *Delete* button on your keyboard.
+
+To **add a new layer**, [add a new tool](#adding-new-tools) in the *Tools* tab.
+
+In the lower left corner of each layer, an icon denotes the layer type: an *icon*, *text*, *rectangle*, or *ellipse* layer.
+
+## Adding new tools
+
+In the *Tools* tab, you can add new layers to the template. Each tool you add to the template equals a new layer. You can add the following tools:
```

Comment:
Feel like there is a difference between tools and layers. Tools are the things you use to add layers. (4 tools are provided to make different layers)

Reply:
Same further in this section.